main.go

```diff
@@ -225,10 +226,15 @@ func addDependencies(_ context.Context, mgr ctrl.Manager, cfg config.Config, v v
 	if err != nil {
 		return fmt.Errorf("failed to start the auto-detect mechanism: %w", err)
 	}
-
+	up := &collectorupgrade.VersionUpgrade{
```

Comment:
this variable is not needed here, it can be moved to the line 237 to minimize the exposure